Add KUTTL make targets for telemetry

paramite opened the pull request:
Depends-On: https://github.com/openstack-k8s-operators/telemetry-operator/pull/213

paramite commented:
recheck

paramite commented:
recheck

paramite commented:
recheck

paramite commented:
recheck

abays commented:
/ok-to-test

abays commented:
recheck

abays commented:
/override data-plane-adoption-github-rdo-centos-9-crc-single-node

abays commented:
/override rdoproject.org/github-check

abays commented on Makefile:
```diff
@@ -2026,6 +2029,19 @@ telemetry_deploy_cleanup: ## cleans up the service instance, Does not affect the
+telemetry_kuttl: export NAMESPACE = ${TELEMETRY_KUTTL_NAMESPACE}
+# Set the value of $TELEMETRY_KUTTL_NAMESPACE if you want to run the telemetry
+# kuttl tests in a namespace different than the default (telemetry-kuttl-tests)
+telemetry_kuttl: input deploy_cleanup telemetry telemetry_deploy_prep
```
If Telemetry operator depends on other operators (such as MariaDB, Keystone, etc), then you'll need to address those dependencies here as pre-targets.  i.e. `kuttl_common_prep`.  For example [1].

[1] https://github.com/openstack-k8s-operators/install_yamls/blob/c0a3767928bc188fcd9af004e977c55b6f2d251d/Makefile#L1611

cescgina commented on Makefile:
```diff
@@ -2026,6 +2029,19 @@ telemetry_deploy_cleanup: ## cleans up the service instance, Does not affect the
+# Set the value of $TELEMETRY_KUTTL_NAMESPACE if you want to run the telemetry
+# kuttl tests in a namespace different than the default (telemetry-kuttl-tests)
+telemetry_kuttl: input deploy_cleanup telemetry telemetry_deploy_prep
+	make wait
```
seems like it's missing ` $(eval $(call vars,$@,telemetry))` before this line to run correctly

cescgina reviewed (commented):
I tried running the tests, but while it finds the `autoscaling` test suite, it does not run any tests because the test files are inside a test folder, but kuttl expects them inside `autoscaling`. So, `path/tests/suites/autoscaling/01-assert.yaml` instead of `path/tests/suites/autoscaling/tests/01-assert.yaml`

paramite commented:
Thanks for hints. Fixed dependencies (added kuttl_common_prep and heat) and added testDirs to config, because telemetry-operator gonna have more than one test suite.

paramite commented:
Currently tests work, just timeouts too early during Prometheus init. Patch in operator [1] is already submitted.

[1] https://github.com/openstack-k8s-operators/telemetry-operator/pull/220

cescgina reviewed (commented):
Changes lgtm, however I could not get the tests to pass unless the namespace fields in the assert file was removed. Looks like the assert is expecting the resources to be deployed in a namespace called `telemetry-kuttl-autoscaling`, apply this commit the tests passed for me: https://github.com/cescgina/telemetry-operator/commit/a2d693c5638fbd91f40d2ae005f7b56a39773e86

paramite commented:
Ah, that's good point. Thanks for that. Do you mind submitting PR with this?

paramite commented:
Your PR is merged. So what now? Do I need to do something else with this PR?

cescgina commented:
> Your PR is merged. So what now? Do I need to do something else with this PR?

PR lgtm, IMO it can be merged now, but I don't have approver rights, @abays can you take a look when free?

vyzigold commented:
/lgtm

fao89 commented:
/approve
